src/handshakestate.rs

```diff
@@ -46,23 +53,109 @@ impl HandshakeState {
         initiator       : bool,
         params          : NoiseParams,
         psks            : [Option<[u8; PSKLEN]>; 10],
+        _kem            : (),
```

Comment:
feels pretty hacktastic this way.

since `kem` is not used in this function and is optional even with the feature flag on, what do you think about having a `HandshakeState::set_kem(&mut self, ...)` function instead and having the builder use that?

in addition to (and because of) this suggestion, i appreciate your moves to refactor this code into subfunctions, but let's in general try to keep this diff as small as possible *especially* in the very sensitive parts of the code like `HandshakeState` and breakup refactors into other pull requests!